lib/validator.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+class Validator
+  VALID_YES_NO_REPLY = /^[yn]$/i.freeze
+
+  def valid_choice?(option)
+    option.match?(/^[1-2]$/)
+  end
+
+  def valid_field?(field, value)
+    {
+      phone: valid_phone?(value),
+      email: valid_email?(value)
+    }.fetch(field, true)
+  end
+
+  def valid_phone?(value)
```

Comment:
`valid_phone?` and `valid_email?` should be private, there is no need to expose them.
The UI only needs to know about `valid_field?`, and nobody else is using these two methods.
Expose as little as you can always.
You can test these two methods through the `valid_field?` method.